src/utils/theme/index.js

```diff
@@ -62,6 +80,19 @@ export const FONTS = {
   comment: {fontFamily: 'Inter-Regular', fontSize: SIZES.body3, lineHeight: 19}
 };
 
+export const hexToRgb = (hex, opacity = 1) => {
+  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+  const r = parseInt(result[1], 16);
+  const g = parseInt(result[2], 16);
+  const b = parseInt(result[3], 16);
+  return result ? `rgba(${r},${g},${b},${opacity})` : null;
+};
+
+export const checkIsHasColor = (color) => {
+  const regex = /^#[0-9a-f]{3,6}$/i;
+  return regex.test(color);
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The newly added utility functions `hexToRgb` and `checkIsHasColor` look fine. However, in the `hexToRgb` function, if the regex execution fails (i.e., the hex color is not valid), it will throw an error when trying to access the result array. To prevent this, you should check if the result is not null before accessing its elements.

```diff
-  const r = parseInt(result[1], 16);
-  const g = parseInt(result[2], 16);
-  const b = parseInt(result[3], 16);
-  return result ? `rgba(${r},${g},${b},${opacity})` : null;
+  if (result) {
+    const r = parseInt(result[1], 16);
+    const g = parseInt(result[2], 16);
+    const b = parseInt(result[3], 16);
+    return `rgba(${r},${g},${b},${opacity})`;
+  }
+  return null;
```

In the `checkIsHasColor` function, the name could be improved for better readability. Something like `isValidHexColor` might be more descriptive.

```diff
- export const checkIsHasColor = (color) => {
+ export const isValidHexColor = (color) => {
   const regex = /^#[0-9a-f]{3,6}$/i;
   return regex.test(color);
 };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->